Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -0,0 +1,76 @@
+class Keg
+  def relocate_dynamic_linkage(relocation)
+    # Patching patchelf using itself fails with "Text file busy" or SIGBUS.
+    return if name == "patchelf"
+    elf_files.each do |file|
+      file.ensure_writable do
+        change_rpath(file, relocation.old_prefix, relocation.new_prefix)
+      end
+    end
+  end
+
+  def change_rpath(file, old_prefix, new_prefix)
+    return unless file.elf? && file.dynamic_elf?
+
+    begin
+      patchelf = Formula["patchelf"].bin/"patchelf"
+    rescue FormulaUnavailableError
+      return
```

Comment:
Seems bad that this will silently succeed

Reply:
I'd rather see a stub or mock in the tests otherwise they aren't going to test this properly.

Reply:
It'll be tested by `brew test-bot` using `testbottest` when it runs `brew bottle testbottest` and `brew install testbottest-0.1.x86_64_linux.bottle.tar.gz`.
https://github.com/Homebrew/homebrew-test-bot/blob/92ae6778765dc124b563d4c169dd0973150273c9/cmd/brew-test-bot.rb#L570

Is that sufficient?

Reply:
> It's only expected to fail with NullLoader used by brew tests.

If it's for failing `brew tests` the relevant tests should have the formula load stubbed or mocked to return a valid object.

> Is that sufficient?

If that's all that's being done on macOS: yes.

Reply:
This still needs addressed to remove the `rescue FormulaUnavailableError.

Reply:
I haven't been able to determine how it is that `brew tests` uses `NullLoader`. I see no reference to `NullLoader` in `test/`. Could you please point me in the right direction?
https://github.com/Homebrew/brew/search?q=NullLoader

Reply:
`HOMEBREW_PREFIX` and so `CoreTap.instance.formula_dir` is set to an empty directory containing no formulae during `brew tests`.

Reply:
I've stubbed out the `patchelf` formula.